### PR TITLE
Glossary and cloud arch updates

### PR DESCRIPTION
**Pull request by Amruta-Ranade**
(Re-doing https://github.com/cockroachdb/docs/pull/18351) 

Addresses https://cockroachlabs.atlassian.net/browse/DOC-9807

In this PR:
- Single-sourced the definition for Request Units
- Updated the [Glossary](https://deploy-preview-18391--cockroachdb-docs.netlify.app/docs/v24.1/architecture/glossary.html) with content for the new tiers
- Deleted the Cloud Arch doc and all references to it as discussed with @marksoper

### CODE REVIEW

File: src/current/_includes/common/basic-terms.md [hunk 1]
```diff
@@ -34,24 +34,28 @@ Service accounts operate under a unified authorization model with organization u
...
 
-#### CockroachDB {{ site.data.products.dedicated }} cluster
+#### CockroachDB {{ site.data.products.standard }} cluster
 
-A CockroachDB {{ site.data.products.cloud }} cluster deployed on request for a specific customer, in a cloud provider's network and compute infrastructure *dedicated* to that customer, which can be distributed over multiple regions for added disaster-resilience. In addition to infrastructure isolation, dedicated clusters can be customized with advanced network, identity-management, and encryption-related security features.
+A CockroachDB {{ site.data.products.cloud }} cluster with full operational features and provisioned capacity, deployed in *shared* network and compute infrastructure.
+
+#### CockroachDB {{ site.data.products.advanced }} cluster
+
+A CockroachDB {{ site.data.products.cloud }} cluster with full operational capacity deployed in a cloud provider's network and compute infrastructure *dedicated* to each customer. In addition to infrastructure isolation, Advanced clusters can be customized with advanced security features for PCI DSS and HIPAA [compliance]({{ link_prefix }}compliance.html) at an additional cost.
```
Comment (kathancox) [inline src/current/_includes/common/basic-terms.md:49]: If the italics on "dedicated" is for emphasis, I think we would want that to be bold. Unless I am missing that this is a term.

File: src/current/_includes/cockroachcloud/request-units.md [hunk 2]
```diff
@@ -0,0 +1 @@
+All cluster activity, including SQL queries, bulk operations, and background jobs, is measured in Request Units, or RUs. An RU is an abstracted metric that represents the [compute and I/O resources]({% link cockroachcloud/resource-usage.md %}#understand-resource-consumption) used by a database operation. In addition to queries that you run, background activity, such as automatic statistics to optimize your queries or connecting a changefeed to an external sink, also consumes RUs. You can see how many request units your cluster has used on the [Cluster Overview]({% link cockroachcloud/cluster-overview-page.md %}#request-units) page.
```
Comment (kathancox) [inline src/current/_includes/cockroachcloud/request-units.md:1]:
```suggestion
All cluster activity, including SQL queries, bulk operations, and background jobs, is measured in Request Units, or RUs. An RU is an abstracted metric that represents the [compute and I/O resources]({% link cockroachcloud/resource-usage.md %}#understand-resource-consumption) used by a database operation. In addition to queries that you run, background activity, such as automatic statistics to optimize your queries or running a changefeed to an external sink, also consumes RUs. You can review how many request units your cluster has used on the [Cluster Overview]({% link cockroachcloud/cluster-overview-page.md %}#request-units) page.
```

File: src/current/_includes/common/basic-terms.md [hunk 3]
```diff
@@ -34,24 +34,28 @@ Service accounts operate under a unified authorization model with organization u
...
 
 #### Request Unit (RU)
 
-Represents the compute and I/O resources used by a query. All database operations in CockroachDB {{ site.data.products.basic }} and {{ site.data.products.standard }} cost a certain amount of RUs depending on the resources used. For example, a "small read" might cost 2 RUs, and a "large read" such as a full table scan with indexes could cost a large number of RUs. You can see how many Request Units your cluster has used on the [Cluster Overview]({{ link_prefix }}cluster-overview-page.html#request-units) page.
+{% include cockroachcloud/request-units.md %}
 
 #### Resource limits
 
-The maximum amounts of storage and RUs a CockroachDB {{ site.data.products.basic }} and {{ site.data.products.standard }} cluster can use in a particular billing period. The amount you are billed is based on the actual resources the cluster used during that billing period.
+The maximum amounts of storage and RUs a CockroachDB {{ site.data.products.basic }} cluster can use in a particular billing period. The amount you are billed is based on the actual resources the cluster used during that billing period.
```
Comment (kathancox) [inline src/current/_includes/common/basic-terms.md:57]: nit: Should "amounts" be plural here? I would expect "The maximum amount of..." But, perhaps there are multiple allotted amounts?

Comment (kathancox) [reply src/current/_includes/common/basic-terms.md:57]: (Also sorry I know you didn't add that, so feel free to ignore this comment if you want!)

File: src/current/_includes/common/basic-terms.md (hunk 1 above)
Comment (Amruta-Ranade) [reply src/current/_includes/common/basic-terms.md:49]: I tried to keep it consistent with the original content (we italicize dedicated and shared in the current copy)

File: src/current/_includes/common/basic-terms.md (hunk 3 above)
Comment (Amruta-Ranade) [inline src/current/_includes/common/basic-terms.md:57]:
```suggestion
The maximum amount of storage and RUs a CockroachDB {{ site.data.products.basic }} cluster can use in a particular billing period. The amount you are billed is based on the actual resources the cluster used during that billing period.
```

**Pull request merged**